docs/docs/firmware/can-traffic.md

```diff
@@ -1,38 +1,46 @@
----
-title: Calculating CAN traffic on a bus
----
 
+# Calculating CAN traffic
 
+In order to calculate the CAN load, we must declare a few variables and their importance within the CAN protocol.
 
-In order to calculate the CAN load, we must declare a few variables and their importance within the CANbus protocol. 
-
-1. Baud: (kbaud)
+## Baud (kbaud)
 
 The baud rate is the rate at which data is transfered in a communication channel (like CAN). For example, 9600 Baud rate is 9600 bits per second
 
-2. Frequency (Hz)
+## Frequency (Hz)
 
-Frequency in the CANbus refers to how many CAN frames (or messages) are transmitted per second. So the greater the frequency, the more bits transferred, increasing the bus load. 
+Frequency is the number of messages transmitted per second for each message type. Different messages can have different frequencies.
 
-3. Total Data Per Cycle
+This variation affects the total bits transferred and, in turn, the bus load. To calculate bits per second, consider each message’s frequency and whether their lengths differ.
 
-Total data per cycle includes all the bits transmitted in one CAN message. This includes not only the payload but overhead bits like start of frame, CRC, acknowledgement, control field etc. 
+## Total Data Per Cycle
```

Comment:
This should be called "Message length".

I would like you to calculate the message length as a function of the data length. You should specify the exact number of bits per message that come from the ID, Start of Frame, Data (dependent on data length), CRC, acknowledge, bit stuffing, end of frame, ex.

Use this site for the details: https://www.csselectronics.com/pages/can-bus-simple-intro-tutorial#can-bus-frames. You should recreate this diagram (or just download and cite it).

It includes all bit lengths BESIDES bit stuffing, which will be dependent on the data length. I'd like you to research what bit stuffing is and the maximum number of bits that it may add to a message.

Your final sentence should be "A message with data length of `x` bytes requires up to `a+bx` bits" where you have calculated `a` and `b`